File: frontend/src/contracts/student.ts
export type AssignmentStatus =
  | "not_started"
  | "in_progress"
  | "remediation"
  | "transfer_ready"
  | "completed";

export type AssignmentSummary = {
  id: string;
  title: string;
  description?: string;
  subject: "math";
  grade: number;
  status: AssignmentStatus;
  progress: {
    completed: number;
    total: number;
  };
  estimatedMinutes?: number;
  assignedAt?: string;
  dueAt?: string;
  nextRoute?: string;
};

export type StudentHomeResponse = {
  student: {
    id: string;
    displayName: string;
    classroomName?: string;
  };
  currentAssignment?: AssignmentSummary;
  recentAssignments: AssignmentSummary[];
};

export type StudentAssignmentsQuery = {
  status?: AssignmentStatus;
  page?: number;
  pageSize?: number;
};

export type StudentAssignmentsResponse = {
  items: AssignmentSummary[];
  page: number;
  pageSize: number;
  totalItems: number;
};

File: backend/app/schemas/student.py
from __future__ import annotations

import uuid
from typing import Literal

from app.schemas.base import ApiSchema


class AssignmentProgressResponse(ApiSchema):
    completed: int
    total: int


class AssignmentSummaryResponse(ApiSchema):
    id: uuid.UUID
    title: str
    description: str | None = None
    subject: Literal["math"] = "math"
    grade: int
    status: str
    progress: AssignmentProgressResponse
    estimated_minutes: int | None = None
    assigned_at: str | None = None
    due_at: str | None = None
    diagnostic_available: bool
    next_route: str | None = None


class StudentHomeStudentResponse(ApiSchema):
    id: uuid.UUID
    display_name: str
    classroom_name: str | None = None
    school_name: str | None = None


class StudentHomeResponse(ApiSchema):
    student: StudentHomeStudentResponse
    current_assignment: AssignmentSummaryResponse | None = None
    recent_assignments: list[AssignmentSummaryResponse]


class StudentAssignmentsResponse(ApiSchema):
    items: list[AssignmentSummaryResponse]
    page: int
    page_size: int
    total: int
